### PR TITLE
Collection improvements

### PR DESCRIPTION
**Pull request by mpurland**

### CODE REVIEW

File: SwiftzTests/ArrayExtSpec.swift [hunk 1]
```diff
@@ -187,6 +187,21 @@ class ArrayExtSpec : XCTestCase {
 
 			return Discard()
 		}
+		
+		property("mapAssociate and mapAssociateLabel behave") <- forAll { (xs : [Int]) in
+			let keyForInt: Int -> String = { "key\($0)" }
+			let dictionary1: [String: Int] = xs.mapAssociate { (keyForInt($0), $0) }
+			let dictionary2: [String: Int] = xs.mapAssociateLabel { keyForInt($0) }
```
Comment (CodaFi) [inline SwiftzTests/ArrayExtSpec.swift]: The usual typelift colons style please!

File: SwiftzTests/ArrayExtSpec.swift [hunk 2]
```diff
@@ -187,6 +187,21 @@ class ArrayExtSpec : XCTestCase {
 
 			return Discard()
 		}
+		
+		property("mapAssociate and mapAssociateLabel behave") <- forAll { (xs : [Int]) in
+			let keyForInt: Int -> String = { "key\($0)" }
```
Comment (CodaFi) [inline SwiftzTests/ArrayExtSpec.swift]: SwiftCheck can generate this function

File: SwiftzTests/ArrayExtSpec.swift (hunk 1 above)
Comment (mpurland) [inline SwiftzTests/ArrayExtSpec.swift]: Sorry about that, probably missed it when I was grabbing it from the other PR.

File: SwiftzTests/ArrayExtSpec.swift (hunk 2 above)
Comment (mpurland) [inline SwiftzTests/ArrayExtSpec.swift]: I'm assuming you mean through the use of ArrowOf<Int, String>?

**Comment by CodaFi** [comment]
Rebase

File: SwiftzTests/DictionaryExtSpec.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    func testProperties() {
+        property("Dictionary init maps key-value pairs from a sequence of that type") <- forAll { (xs : [Int]) in
+            return forAll { (f : ArrowOf<Int, String>) in
+                let pairs = xs.map { (f.getArrow($0), $0) }
+                let dictionary = Dictionary(pairs)
+                return xs.all { dictionary[f.getArrow($0)] == $0 }
+            }
+        }
+    }
+}
```
Comment (CodaFi) [inline SwiftzTests/DictionaryExtSpec.swift]: Newline here

Comment (mpurland) [inline SwiftzTests/DictionaryExtSpec.swift]: :white_check_mark:

**Comment by CodaFi** [comment]
:boat:

**Pull request merged**